packages/aws-rfdk/lib/deadline/scripts/bash/configureWorkerHealthCheck.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script configures the deadline.ini file with the settings for the HealthMonitor
+# Arguments:
+# $1: healthCheckPort
+# $2: minimum supported deadline version
+
+# exit when any command fails
+set -xeuo pipefail
+
+HEALTH_CHECK_PORT="$1"
+MINIMUM_SUPPORTED_DEADLINE_VERSION=$2
+
+# Cloud-init does not load system environment variables. Cherry-pick the
+# environment variable installed by the Deadline Client installer.
+if [ -f "/etc/profile.d/deadlineclient.sh" ]; then
+    source "/etc/profile.d/deadlineclient.sh"
+fi
+
+# Find Deadline
+if [ -z "$DEADLINE_PATH" ]; then
+    echo "ERROR: DEADLINE_PATH environment variable not set"
+    exit 1
+fi
+
+DEADLINE_COMMAND="$DEADLINE_PATH/deadlinecommand"
+if [ ! -f "$DEADLINE_COMMAND" ]; then
+    echo "ERROR: $DEADLINE_COMMAND not found!"
+    exit 1
+fi
+
+isVersionLessThan() {
+    python -c "import sys;sys.exit(0 if tuple(map(int, sys.argv[-2].split('.'))) < tuple(map(int, sys.argv[-1].split('.'))) else 1)" "$1" "$2"
+}
+
+DEADLINE_VERSION=$("$DEADLINE_COMMAND" -Version | grep -oP '[v]\K\d+\.\d+\.\d+\.\d+\b')
+if [ -z "$DEADLINE_VERSION" ]; then
+    echo "ERROR: Unable to identify the version of installed Deadline Client. Exiting..."
+    exit 1
+fi
+
+if isVersionLessThan $DEADLINE_VERSION $MINIMUM_SUPPORTED_DEADLINE_VERSION; then
+    echo "ERROR: Installed Deadline Version ($DEADLINE_VERSION) is less than the minimum supported version ($MINIMUM_SUPPORTED_DEADLINE_VERSION). Exiting..."
+    exit 1
+fi
+
+# enabling the health check port
+"$DEADLINE_COMMAND" -SetIniFileSetting ResourceTrackerVersion V2
+# health check port
+"$DEADLINE_COMMAND" -SetIniFileSetting LauncherHealthCheckPort $HEALTH_CHECK_PORT
+
+# Restart service, if it exists, else restart application
+if service --status-all | grep -q 'Deadline 10 Launcher'; then
+  service deadline10launcher restart
+else
+  DEADLINE_LAUNCHER="$DEADLINE_PATH/deadlinelauncher"
+  "$DEADLINE_LAUNCHER" -shutdownall
+  "$DEADLINE_LAUNCHER"
+fi
```

Comment:
In the `WorkerInstanceFleet` case, we will now be restarting the launcher and Worker twice at startup. I understand it is necessary to ensure the changes take effect when running this script in isolation, but to avoid this extra restart, we could use OO-inheritance of `WorkerConfiguration`, like this:

```ts
class WorkerConfiguration {
  private readonly applyConfigScript: ScriptAsset;
  private readonly setConfigScript: ScriptAsset;

  constructor(...) {
    super(...);

    // Script to restart the launcher
    this.applyConfigScript = ScriptAsset.fromPathConvention(...);

    this.setConfiguration();
    this.applyConfiguration();
  }

  protected setConfiguration() {
    setConfigScript.executeOn(...);
  }

  protected applyConfiguration() {
    // Calls a separate ScriptAsset to restart the launcher
    applyConfigScript.executeOn(...);
  }
}
```

and for the worker instance fleet we override the `setConfiguration` method:

```ts
class WorkerInstanceFleetConfiguration extends WorkerConfiguration {
  private readonly configureHealthMonitor: ScriptAsset;

  constructor(...) {
    super(...);
    this.configureHealthMonitor = ScriptAsset.fromPathConvention(...);
  }

  protected setConfiguration() {
    super.setConfiguration();
    this.configureHealthMonitor.executeOn(...);
  }
}
```

This assumes my other proposed change of applying the configuration in the constructor. It's a bit of rework, but do you think it's worth it?